src/fmeval/eval_algorithms/general_semantic_robustness.py

```diff
@@ -220,26 +158,13 @@ def build_pipeline(
             multiple inputs from the dataset are used.
         :returns: A TransformPipeline that can be used by either `evaluate_sample` or `evaluate`.
         """
-        get_perturbed_inputs = self.perturbation_transform
-
-        # Generate prompts from perturbed inputs
-        gen_perturbed_prompts = GeneratePrompt(
-            input_keys=get_perturbed_inputs.output_keys,
-            output_keys=[
-                create_output_key(GeneratePrompt.__name__, perturbed_input_key)
-                for perturbed_input_key in get_perturbed_inputs.output_keys
-            ],
-            prompt_template=prompt_template,
-        )
 
-        # Invoke model with prompts generated above
-        get_perturbed_responses = GetModelOutputs(
-            input_to_output_keys={
-                perturbed_prompt_key: [create_output_key(GetModelOutputs.__name__, perturbed_prompt_key)]
-                for perturbed_prompt_key in gen_perturbed_prompts.output_keys
-            },
-            model_runner=model,
+        transforms = get_model_responses_from_perturbed_inputs(
```

Comment:
why not merge 162 and 167, why do we need variable transforms here?

Reply:
No good reason; I can merge them for greater clarity.